app/navigation/LoadingNavigator.scala

```diff
@@ -29,7 +30,15 @@ class LoadingNavigator {
     case AddUnLocodeYesNoPage         => ua => addUnlocodeNormalRoute(ua, departureId)
     case UnLocodePage                 => ua => AddExtraInformationYesNoPage.route(ua, departureId, NormalMode)
     case AddExtraInformationYesNoPage => ua => addExtraInformationYesNoNormalRoute(ua, departureId)
-    case CountryPage                  => ua => LocationPage.route(ua, departureId, NormalMode)
+    case CountryPage =>
+      ua =>
+        if (ua.departureData.isSimplified) {
+          ua.departureData.TransitOperation.limitDate match {
+            case Some(_) => LocationPage.route(ua, departureId, NormalMode)
+            case None    => LimitDatePage.route(ua, departureId, mode)
+          }
+        } else
+          LocationPage.route(ua, departureId, NormalMode)
```

Comment:
small refactor but could this be:
`if (ua.departureData.isSimplified | ua.departureData.TransitOperation.limitDate.isDefined) .... else ....`